packages/contracts/src/viemClient.ts

```diff
@@ -0,0 +1,131 @@
+  chain: arbitrumNova,
+  transport: fallback([
+    http(process.env.REACT_APP_ARBITRUM_NOVA_NODE_URL),
+    // https://github.com/DefiLlama/chainlist/blob/83b8cc32ee79c10e0281e1799ebe4cd1696082b7/constants/extraRpcs.js#L1393
+    http('https://nova.arbitrum.io/rpc'),
+  ]),
+}) as PublicClient
+
+export const viemOptimismClient = createPublicClient({
+  chain: optimism,
+  transport: fallback([
+    http(process.env.REACT_APP_OPTIMISM_NODE_URL),
+    // https://github.com/DefiLlama/chainlist/blob/83b8cc32ee79c10e0281e1799ebe4cd1696082b7/constants/llamaNodesRpcs.js#L41
+    http('https://optimism.llamarpc.com'),
+  ]),
+}) as PublicClient
+
+export const viemGnosisClient = createPublicClient({
+  chain: gnosis,
+  transport: fallback([
+    // https://github.com/DefiLlama/chainlist/blob/83b8cc32ee79c10e0281e1799ebe4cd1696082b7/constants/extraRpcs.js#L1978
+    http(process.env.REACT_APP_GNOSIS_NODE_URL),
+    http('https://rpc.gnosischain.com'),
+  ]),
+}) as PublicClient
+
+export const viemPolygonClient = createPublicClient({
+  chain: polygon,
+  transport: fallback([
+    http(process.env.REACT_APP_POLYGON_NODE_URL),
+    // https://github.com/DefiLlama/chainlist/blob/83b8cc32ee79c10e0281e1799ebe4cd1696082b7/constants/llamaNodesRpcs.js#L52
+    http('https://polygon.llamarpc.com'),
+  ]),
+}) as PublicClient
+
+export const viemBaseClient = createPublicClient({
+  chain: base,
+  transport: fallback([
+    http(process.env.REACT_APP_BASE_NODE_URL),
+    // https://github.com/DefiLlama/chainlist/blob/83b8cc32ee79c10e0281e1799ebe4cd1696082b7/constants/llamaNodesRpcs.js#L19
+    http('https://base.llamarpc.com'),
+  ]),
+}) as PublicClient
+
+export const viemClientByChainId: Record<ChainId, PublicClient> = {
+  [KnownChainIds.EthereumMainnet]: viemEthMainnetClient,
+  [KnownChainIds.BnbSmartChainMainnet]: viemBscClient,
+  [KnownChainIds.AvalancheMainnet]: viemAvalancheClient,
+  [KnownChainIds.ArbitrumMainnet]: viemArbitrumClient,
+  [KnownChainIds.ArbitrumNovaMainnet]: viemArbitrumNovaClient,
+  [KnownChainIds.GnosisMainnet]: viemGnosisClient,
+  [KnownChainIds.PolygonMainnet]: viemPolygonClient,
+  // cast required for these due to typescript shenanigans
+  // https://github.com/wagmi-dev/viem/issues/1018
+  [KnownChainIds.OptimismMainnet]: viemOptimismClient,
+  [KnownChainIds.BaseMainnet]: viemBaseClient,
+}
+
+export const viemClientByNetworkId: Record<number, PublicClient> = {
+  [mainnet.id]: viemEthMainnetClient,
+  [bsc.id]: viemBscClient,
+  [avalanche.id]: viemAvalancheClient,
+  [arbitrum.id]: viemArbitrumClient,
+  [arbitrumNova.id]: viemArbitrumNovaClient,
+  [gnosis.id]: viemGnosisClient,
+  [polygon.id]: viemPolygonClient,
+  // cast required for these due to typescript shenanigans
+  // https://github.com/wagmi-dev/viem/issues/1018
+  [optimism.id]: viemOptimismClient,
+  [base.id]: viemBaseClient,
+}
+
+export const assertGetViemClient = (chainId: ChainId): PublicClient => {
```

Comment:
we should remove `src/lib/viem-client` and replace usage with contracts package

Reply:
hell yeah, right here sir
dc4ba3b663117fb0cf3a89930c2bbf6f6d278042